test/cppsuite/test_harness/runtime_monitor.cxx

```diff
@@ -338,4 +351,30 @@ runtime_monitor::finish()
           "limits.");
 }
 
+/*
+ * This function generates a file that contains the values of the different statistics that need to
+ * be saved as indicated by the configuration file.
+ */
+void
+runtime_monitor::save_stats(const std::string &filename)
+{
+    std::string stat_info;
+    std::ofstream file(filename + ".json");
+    file << "[{\"info\":{\"test_name\": \"" << filename << "\"},\"metrics\": [";
+
+    for (const auto &stat : _stats) {
+        if (stat->get_save())
+            stat_info += "{\"name\":\"" + stat->get_name() +
```

Comment:
Monir note: why do we need `stat_info` if we already opened `file` stream and started pushing data in it? We can either use the string buffer or the file stream for all the data.

Reply:
We could use the file stream directly indeed. But the annoying part is the extra comma at the end that is removed lin 373. That's why I wanted a temporary `string` so I can edit it before writing it into the file.

Reply:
You could use the string buffer to build JSON, create the file stream and flush the data at the very end. Or use the stream for everything add logic to add a comma at the beginning of the `for` loop.

Reply:
I used a `string` for the entire buffer before adding it to the file, see 5be76e2d74b355999f0dc6bbd98790ea3ba8bb5b. Thanks for this good suggestion!